diff --git a/src/vpc_gw/vpc_gw.c b/src/vpc_gw/vpc_gw.c
index 966994c..420a2bc 100644
--- a/src/vpc_gw/vpc_gw.c
+++ b/src/vpc_gw/vpc_gw.c
@@ -9,13 +9,13 @@
 /* Includes */
 #include "vpc_gw.h"
 
+#include "diagnostics/diagnostics_wrapper.h"
 #include "interface_abstraction/digital_output_writer.h"
 #include "logger.h"
 #include "pal_can.h"
 #include "pal_data_dictionary.h"
 #include "pal_digital_io.h"
 #include "power_state_manager/power_state_manager.h"
-#include "diagnostics/diagnostics_wrapper.h"
 #include "vpc_gw_data_dictionary.h"
 
 #if defined(BACKEND_ENGINE_PIINNOVO_M130)
@@ -29,7 +29,7 @@
 #endif
 #else
 /* Define calibratable for NVM, necessary for SIL */
-PAL_NVM_CAL uint8_t palv_nvm_default_data[1];
+extern PAL_NVM_CAL uint8_t palv_nvm_default_data[1];
 #endif
 
 /* Defines and Type definitions */
diff --git a/src/vpc_scu/pal_main.c b/src/vpc_scu/pal_main.c
index 119edbd..959bef0 100644
--- a/src/vpc_scu/pal_main.c
+++ b/src/vpc_scu/pal_main.c
@@ -22,7 +22,7 @@
 #endif
 #else
 /* Define calibratable for NVM, necessary for SIL */
-PAL_NVM_CAL uint8_t palv_nvm_default_data[1];
+extern PAL_NVM_CAL uint8_t palv_nvm_default_data[1];
 #endif
 
 #ifndef ENABLE_SECONDARY_PROC
diff --git a/src/vpc_vcu/vpc_vcu.c b/src/vpc_vcu/vpc_vcu.c
index 26d8f79..00b03e6 100644
--- a/src/vpc_vcu/vpc_vcu.c
+++ b/src/vpc_vcu/vpc_vcu.c
@@ -9,13 +9,13 @@
 /* Includes */
 #include "vpc_vcu.h"
 
+#include "diagnostics/diagnostics_wrapper.h"
 #include "interface_abstraction/digital_output_writer.h"
 #include "logger.h"
 #include "pal_can.h"
 #include "pal_data_dictionary.h"
 #include "pal_digital_io.h"
 #include "vpc_vcu/power_state_controller/power_state_controller.h"
-#include "diagnostics/diagnostics_wrapper.h"
 #include "vpc_vcu_data_dictionary.h"
 
 /* NVM usage */
@@ -30,7 +30,7 @@
 #endif
 #else
 /* Define calibratable for NVM, necessary for SIL */
-PAL_NVM_CAL uint8_t palv_nvm_default_data[1];
+extern PAL_NVM_CAL uint8_t palv_nvm_default_data[1];
 #endif
 
 /* Defines and Type definitions */
